controllers/commentController.js

```diff
@@ -148,7 +140,161 @@ export default class CommentController {
+      });
+    }
+  }
+
+  /**
+   * @description Allows users to fetch a particular comment of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+  static async fetch(req, res) {
+    try {
+      const { id: commentId } = req.params;
+      const comment = await Comment.findOne({ where: { id: commentId } });
+
+      if (!comment) {
+        return res
+          .status(BAD_REQUEST)
+          .json({ message: "The comment does not exist" });
+      }
+      res.status(OK).json({ comment });
+    } catch (error) {
+      res.status(INTERNAL_SERVER_ERROR).json({ message: error.message });
+    }
+  }
+
+  /**
+   * @description Allows users to fetch all comments of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+  static async fetchAllComments(req, res) {
+    try {
+      const { slug } = req.params;
+
+      const article = await Article.findOne({
+        where: { slug },
+        include: [
+          {
+            model: Comment,
+            attributes: ["body"],
+            include: [{ model: User, as: "author", attributes: ["username"] }]
+          }
+        ]
+      });
+
+      if (!article) {
+        return res
+          .status(BAD_REQUEST)
+          .json({ message: "There is no article with that slug" });
+      }
+      res.status(OK).json({ article });
+    } catch (error) {
+      res.status(INTERNAL_SERVER_ERROR).json({ message: error.message });
+    }
+  }
+  /**
+   * @description Allows users to update a comment of an article.
+   * @param  {Object} req - The request object.
+   * @param  {Object} res - The response object.
+   * @returns {Object} - It returns the request response object.
+   */
+
+  static async update(req, res) {
+    try {
+      const { body: newBody } = req.body;
+      const { id: commentId } = req.params;
+
+      if (newBody === " ") {
+        return res.status(BAD_REQUEST).json({ message: "body should not be empty" });
+      }
+      const oldVersion = await Comment.findOne({
+        where: { id: commentId },
+        attributes: ["body"]
+      });
+      const comment = await Comment.update(
+        { body: newBody },
+        { row: true, returning: true, where: { id: commentId } }
+      );
+
+      if (comment[0] === 0) {
+        return res
+          .status(BAD_REQUEST)
+          .json({ message: "The comment does not exist" });
+      }
+      const commentHistory = await Commentlog.create({
+        old_version: oldVersion.body,
+        new_version: newBody,
+        commentId: comment[1][0].dataValues.id
+      });
+
+      res.status(CREATED).json({
+        message: "Comment updated successfully",
+        comment: comment[1][0].dataValues,
+        commentHistory: commentHistory.dataValues
```

Comment:
@YvesIraguha Why are you returning the new comment twice, as comment and in commentHistory? Can't you just return the old and new version?